addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/handler/BridgeV3Handler.java

```diff
@@ -85,12 +88,13 @@ protected void startConnectAndKeepAlive() {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
         }
 
-        new Thread(this::receive).start();
-        running = true;
+        running = scheduler.scheduleWithFixedDelay(this::receive, 0, config.refreshTime, TimeUnit.SECONDS);
     }
 
     protected void stopKeepAlive() {
-        running = false;
+        if (running != null) {
+            running.cancel(false);
```

Comment:
Could you also set `running` to `null` here, to allow the garbage collector to collect this object?

Reply:
Done